sdl/v2.go

```diff
@@ -24,11 +52,92 @@ type v2ExposeTo struct {
 }
 
 type v2Expose struct {
-	Port   uint16
-	As     uint16
-	Proto  string       `yaml:"proto,omitempty"`
-	To     []v2ExposeTo `yaml:"to,omitempty"`
-	Accept v2Accept     `yaml:"accept"`
+	Port        uint16
+	As          uint16
+	Proto       string        `yaml:"proto,omitempty"`
+	To          []v2ExposeTo  `yaml:"to,omitempty"`
+	Accept      v2Accept      `yaml:"accept"`
+	HTTPOptions v2HTTPOptions `yaml:"http_options"`
+}
+
+type v2HTTPOptions struct {
+	MaxBodySize uint32   `yaml:"max_body_size"`
+	ReadTimeout uint32   `yaml:"read_timeout"`
+	SendTimeout uint32   `yaml:"send_timeout"`
+	NextTries   uint32   `yaml:"next_tries"`
+	NextTimeout uint32   `yaml:"next_timeout"`
+	NextCases   []string `yaml:"next_cases"`
+}
+
+func (ho v2HTTPOptions) any() bool {
+	return len(ho.NextCases) != 0 ||
+		ho.MaxBodySize != 0 ||
+		ho.ReadTimeout != 0 ||
+		ho.SendTimeout != 0 ||
+		ho.NextTries != 0 ||
+		ho.NextTimeout != 0
+}
+
+func (ho v2HTTPOptions) asManifest() (manifest.ServiceExposeHTTPOptions, error) {
+	maxBodySize := ho.MaxBodySize
+	if 0 == maxBodySize {
+		maxBodySize = defaultMaxBodySize
+	} else if maxBodySize > upperLimitBodySize {
+		return manifest.ServiceExposeHTTPOptions{}, fmt.Errorf("%w: body size cannot be greater than %d bytes", errHTTPOptionNotAllowed, upperLimitBodySize)
+	}
+
+	readTimeout := ho.ReadTimeout
+	if 0 == readTimeout {
+		readTimeout = defaultReadTimeout
+	} else if readTimeout > upperLimitReadTimeout {
+		return manifest.ServiceExposeHTTPOptions{}, fmt.Errorf("%w: read timeout cannot be greater than %d ms", errHTTPOptionNotAllowed, upperLimitReadTimeout)
+	}
+
+	sendTimeout := ho.SendTimeout
+	if 0 == sendTimeout {
+		sendTimeout = defaultSendTimeout
+	} else if sendTimeout > upperLimitSendTimeout {
+		return manifest.ServiceExposeHTTPOptions{}, fmt.Errorf("%w: send timeout cannot be greater than %d ms", errHTTPOptionNotAllowed, upperLimitSendTimeout)
+	}
+
+	nextTries := ho.NextTries
+	if 0 == nextTries {
+		nextTries = defaultNextTries
+	}
+
+	nextCases := ho.NextCases
+	if len(nextCases) == 0 {
+		nextCases = defaultNextCases
+	} else {
+		for _, nextCase := range nextCases {
+			switch nextCase {
+			case nextCaseOff:
+				if len(nextCases) != 1 {
+					return manifest.ServiceExposeHTTPOptions{}, errCannotSpecifyOffAndOtherCases
+				}
+			case nextCaseError:
+			case nextCaseTimeout:
```

Comment:
missing `fallthrough` ?

Reply:
It should take no action in the event that this case is hit, which is the correct thing